sense-android-library/src/nl/sense_os/service/storage/DbHelper.java

```diff
@@ -41,22 +54,63 @@
     public DbHelper(Context context, boolean persistent) {
         // if the database name is null, it will be created in-memory
         super(context, persistent ? DATABASE_NAME : null, null, DATABASE_VERSION);
+
+        SharedPreferences pref = context.getSharedPreferences(SensePrefs.MAIN_PREFS,
+                Context.MODE_PRIVATE);
+        boolean encrypt = pref.getBoolean(Advanced.ENCRYPT_SENSOR_DATA, false);
+
+        if (encrypt) {
+            TelephonyManager telephonyManager = (TelephonyManager) context.getSystemService(Context.TELEPHONY_SERVICE);
+            String imei = telephonyManager.getDeviceId();
+
+            setPassphrase(imei);
+        }
+
+        SQLiteDatabase.loadLibs(context);
+
+        // check for old plain database
+        if (persistent && encrypt) {
+            // try to open the database with the password from the user.
+            // The only possible error now must be a wrong password or using plain database.
+            try {
+                getWritableDatabase(passphrase);
+            } catch (SQLiteException e) {
+                Log.v(TAG, "Encrypting old database with passphrase "+passphrase);
```

Comment:
@orhan89 This should be removed, because if you give the password then we don't have to encyrpt it at all